code-reviews/pull-requests.md

```diff
@@ -33,17 +33,31 @@ We should always aim to have pull requests be as small as possible, without losi
 1. They are easier to deploy; this is aligned with the strategy of release fast and release often.
 1. Minimizes possible conflicts and stale PRs, which are difficult to merge and keep in sync with master either because they're very dynamic or contain refactoring.
 
-However, we should avoid having PRs that include code that is without context or loosely coupled.
+However, we should avoid having PRs that include code that is without context or loosely coupled. There is no right size, but keep in mind that a code review is a collaborative process, and you should work with your reviewer to find out a reasonable size. Besides the size, remember that every PR should:
 
-<!-- TODO: Expand the topics below
+1. Be consistent and within context, 
+1. not break the build, and
+1. include related tests (UTs, integration, etc.) as part of the PR.
+ 
+There are times where seems a big PR is unavoidable and necessary; the act of breaking it up would be more time consuming and error-prone than it would be beneficial, and that it is the reason why it's easier to write small PRs in the first place.
 
-There are times where seem a big PR is unavoidable; however, there are some strategies to keep PRs small depending on the "cause" of the ineluctability:
+These are some strategies to keep PRs small depending on the "cause" of the ineluctability:
 
 ### Minimum Working Components
+
+If the design of your changes allows breaking the implementation on working chunks, self-contained changes, that do not break the build, it is better to create one PR per component. The components should be fully functional pieces that can be released as soon as possible. Think about this as breaking the implementation vertically, where components sometimes involve changes on the full-stack.
```

Comment:
breaking the implementation on working chunks, self-contained changes -> breaking the implementation into self-contained changes that do not break the build, ...